### ISSUE TITLE
Query builder WHERE does not convert != to NOT NULL in case if parameter value is null

### ISSUE THREAD

**Issue by Arkemlar** [doctrine/doctrine2#7289]
### Bug Report

|    Q        |   A
|------------ | ------
| BC Break    | dont know
| Version     | v2.6.1 (doctrine/orm)

#### Summary

Using MYSQL. Imagine simple query:
```
$qb
  ->andWhere('entity.id != :value')
  ->setParameter('value', $someValue);
```
If `$someValue` is int - then OK, but if NULL - then query executes and returns 0 rows. Thats because doctrine builds query like `WHERE entity.id <> NULL`. I expected that `!=` will be converted to `IS NOT`.

#### Current behavior

`!=` converts to `<>` if NULL passed as parameter value.

#### How to reproduce

```
// init some query builder
$qb
  ->andWhere('entity.id != :value')
  ->setParameter('value', null);
// execute query and check generated SQL
```

#### Expected behavior

`!=` converts to `IS NOT` if NULL passed as parameter value.

**Comment by parijke** [comment]
How to handle this situation? I ran into the same issue, where a parameter can be null. but != null obviousely returns the worng result?

**Comment by Adamko23** [comment]
omg, really is it that hard make the conversion in doctrine? Cause programmer must do this disgusting thing

```
if(is_null($value)){
    $qb->andWhere('entity.id IS NULL');
} else { 
    $qb->andWhere('entity.id != :value')->setParameter('value', $value);
}
```

**Comment by Ocramius** [comment]
Yes, it is part of the SQL standard: `IS NULL` is the correct way to do this. 

Setting parameter values also doesn't modify the SQL/DQL structure pre-defined before: it's too late to modify a prepared statement while parameters are being bound to it. 

 > omg, really is it that hard make the conversion in doctrine?

Yes, really.
